Views/VMDisplayMetalViewController+Pointer.m

```diff
@@ -0,0 +1,116 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "VMDisplayMetalViewController.h"
+#import "VMDisplayMetalViewController+Touch.h"
+#import "VMDisplayMetalViewController+Pointer.h"
+#import "VMCursor.h"
+#import "CSDisplayMetal.h"
+
+NS_AVAILABLE_IOS(13.4)
+@implementation VMDisplayMetalViewController (Pointer)
+
+- (void)addInteractionTo:(UIView*)view {
+    UIPointerInteraction *pointerInteraction = [[UIPointerInteraction alloc] initWithDelegate:self];
+    [view addInteraction:pointerInteraction];
+}
+
+// Add pointer interaction all interactive subviews except the keyboard
+-(void)initPointerInteraction {
+    [self addInteractionTo:self.mtkView];
+    [self addInteractionTo:self.powerExitButton];
+    [self addInteractionTo:self.pauseResumeButton];
+    [self addInteractionTo:self.restartButton];
+    [self addInteractionTo:self.zoomButton];
+    [self addInteractionTo:self.toggleKeyboardButton];
+    [self addInteractionTo:self.configureButton];
+    [self addInteractionTo:self.hideToolbarButton];
+    [self addInteractionTo:self.resumeBigButton];
+}
+
+#pragma mark - UIPointerInteractionDelegate
+- (UIPointerStyle *)pointerInteraction:(UIPointerInteraction *)interaction styleForRegion:(UIPointerRegion *)region {
+    // Hide cursor while hovering in VM view
+    if (interaction.view == self.mtkView) {
+        return [UIPointerStyle hiddenPointerStyle];
+    } else {
+        return [UIPointerStyle styleWithShape:[UIPointerShape shapeWithRoundedRect:interaction.view.bounds] constrainedAxes:UIAxisBoth];
+    }
+}
+
+static CGFloat CGPointToPixel(CGFloat point) {
+    return point * [UIScreen mainScreen].scale; // FIXME: multiple screens?
+}
+
+- (bool)isPointOnVMDisplay:(CGPoint)pos {
+    CGSize screenSize = self.mtkView.drawableSize;
+    CGSize scaledSize = {
+        self.vmDisplay.displaySize.width * self.vmDisplay.viewportScale,
+        self.vmDisplay.displaySize.height * self.vmDisplay.viewportScale
+    };
+    CGRect drawRect = CGRectMake(
+        self.vmDisplay.viewportOrigin.x + screenSize.width/2 - scaledSize.width/2,
+        self.vmDisplay.viewportOrigin.y + screenSize.height/2 - scaledSize.height/2,
+        scaledSize.width,
+        scaledSize.height
+    );
+    pos.x -= drawRect.origin.x;
+    pos.y -= drawRect.origin.y;
+    if (pos.x < 0) {
+        return false;
+    } else if (pos.x > scaledSize.width) {
+        return false;
+    }
+    if (pos.y < 0) {
+        return false;
+    } else if (pos.y > scaledSize.height) {
+        return false;
+    }
+    return true;
```

Comment:
If not the above, `return 0 <= pos.x && pos.x <= scaledSize.width && 0 <= pos.y && pos.y <= scaledSize.height`?

Reply:
That works, thank you.